src/tests/functional/navigation_tests.js

```diff
@@ -539,3 +540,54 @@ test("ignores forms with a [target] attribute that target an iframe with [name='
 
   await expect(page).toHaveURL(withPathname("/src/tests/fixtures/one.html"))
 })
+
+test("resets scroll position on 404 error with turbo-refresh-scroll=reset", async ({ page }) => {
+  await page.goto("/src/tests/fixtures/navigation_scroll_reset.html")
+  await page.evaluate(() => window.scrollTo(0, 100))
+  assert.notOk(await isScrolledToTop(page), "page is scrolled down")
+
+  await page.click("#link-404")
+  await nextBody(page)
+  await nextBeat()
+
+  assert.ok(await isScrolledToTop(page), "page is scrolled to the top after 404 error")
+  assert.equal(await page.locator("h1").textContent(), "Not Found")
```

Comment:
If these `assert` calls were replaced with async `expect`-based expectations, and the `h1` assertions were first, the `nextBody` and `nextBeat` calls might not be necessary:

Could these calls to `assert` be replaced with `expect` calls?

```suggestion
  expect(await isScrolledToTop(page), "page is scrolled down").toEqual(false)

  await page.click("#link-404")

  await expect(page.locator("h1")).toHaveText("Not Found")
  expect(await isScrolledToTop(page), "page is scrolled to the top after 404 error").toEqual(true)
})
```